mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -689,6 +691,21 @@ public List<JobArtifact> listJobArtifacts(String name, String version) throws IO
             }).filter(Objects::nonNull).collect(Collectors.toList());
     }
 
+    @Override
+    public void addNewJobArtifactToCache(ClusterID clusterID, ArtifactID artifactID) throws IOException {
+        kvStore.upsert(
+            JOB_ARTIFACTS_TO_CACHE_PER_CLUSTER_ID_NS,
+            clusterID.getResourceID(),
+            clusterID.getResourceID(),
```

Comment:
It seems that only one artifact can be stored per resource cluster. Can you please confirm if this is the case and, if so, modify it to allow for multiple artifacts to be cached per resource cluster? Thank you.